portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -1099,6 +1118,61 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 	    });
 	    return def.promise();
 	},
+	// make new functions for heatmap to bypass OQL filters and handle continuous data
+	//
+	'getHeatmapData': function (genetic_profile_id, genes, sample_or_patient) {
+	    var def = new $.Deferred();
+	    var self = this;
+	    var sample_ids = self.getSampleIds();
+	    var genes = genes || [];
+	    var sample_or_patient = sample_or_patient || "sample";
+	    $.when(window.cbioportal_client.getGeneticProfileDataBySample({
+		'genetic_profile_ids': [genetic_profile_id],
+		'genes': genes.map(function(x) { return x.toUpperCase(); }),
+		'sample_ids': sample_ids
+	    }), self.getPatientSampleIdMap()
+	    ).then(function (client_sample_data, sample_to_patient_map) {
+		var interim_data = {};
+		for (var i = 0; i < genes.length; i++) {
+		    var gene = genes[i].toUpperCase();
+		    interim_data[gene] = {};
+		    for (var j = 0; j < sample_ids.length; j++) {
+			var id = sample_ids[j];
+			interim_data[gene][id] = {};
+			interim_data[gene][id]["hugo_gene_symbol"] = gene;
+			interim_data[gene][id][sample_or_patient] = (sample_or_patient === "patient" ? sample_to_patient_map[id] : id);
```

Comment:
@adamabeshouse, what did the Oncoprint previously do when `setTrackData()` was given a list with multiple data for the same patient?